src/react-components/room/PeopleSidebar.js

```diff
@@ -0,0 +1,141 @@
+import { ReactComponent as VolumeHighIcon } from "../icons/VolumeHigh.svg";
+import { ReactComponent as VolumeMutedIcon } from "../icons/VolumeMuted.svg";
+import { List, ButtonListItem } from "../layout/List";
+
+function getDeviceLabel(ctx) {
+  if (ctx) {
+    if (ctx.mobile) {
+      return "On Mobile";
+    } else if (ctx.discord) {
+      return "Discord Bot";
+    } else if (ctx.hmd) {
+      return "In VR";
+    }
+  }
+
+  return "On Desktop";
+}
+
+function getDeviceIconComponent(ctx) {
+  if (ctx) {
+    if (ctx.mobile) {
+      return PhoneIcon;
+    } else if (ctx.discord) {
+      return DiscordIcon;
+    } else if (ctx.hmd) {
+      return VRIcon;
+    }
+  }
+
+  return DesktopIcon;
+}
+
+function getVoiceLabel(micPresence) {
+  if (micPresence) {
+    if (micPresence.talking) {
+      return "Talking";
+    } else if (micPresence.muted) {
+      return "Muted";
+    }
+  }
+
+  return "Not Talking";
+}
+
+function getVoiceIconComponent(micPresence) {
+  if (micPresence) {
+    if (micPresence.talking) {
+      return VolumeHighIcon;
+    } else if (micPresence.muted) {
+      return VolumeMutedIcon;
+    }
+  }
+
+  return VolumeOffIcon;
+}
+
+// TODO: i18n
+function getPresenceMessage(presence) {
+  switch (presence) {
+    case "lobby":
+      return "In Lobby";
+    case "room":
+      return "In Room";
+    case "entering":
+      return "Entering Room";
+    default:
+      return undefined;
+  }
+}
+
+function getPersonName(person) {
+  return person.profile.displayName + (person.isMe ? " (You)" : "");
+}
+
+function getLabel(person) {
+  if (person.context.discord) {
+    return `${getDeviceLabel(person.context)}, ${getPersonName(person)} is ${getPresenceMessage(person.presence)}.`;
+  }
+
+  return `${person.roles.owner ? "Moderator " : ""}${getPersonName(person)}, is ${getPresenceMessage(
+    person.presence
+  )} and ${getVoiceLabel(person.micPresence)} ${getDeviceLabel(person.context)}.`;
+}
+
+export function PeopleSidebar({ people, onSelectPerson, onClose, showMuteAll, onMuteAll }) {
+  return (
+    <Sidebar
+      title={`People (${people.length})`}
+      beforeTitle={<CloseButton onClick={onClose} />}
+      afterTitle={showMuteAll ? <SidebarButton onClick={onMuteAll}>Mute All</SidebarButton> : undefined}
+    >
+      <List>
+        {people.map(person => {
+          const DeviceIcon = getDeviceIconComponent(person.context);
+          const VoiceIcon = getVoiceIconComponent(person.micPresence);
+
+          return (
+            <ButtonListItem
+              className={styles.person}
+              key={person.id}
+              type="button"
+              aria-label={getLabel(person)}
+              onClick={e => onSelectPerson(person, e)}
+            >
+              {<DeviceIcon title={getDeviceLabel(person.context)} />}
+              {!person.context.discord && VoiceIcon && <VoiceIcon title={getVoiceLabel(person.micPresence)} />}
+              <p>{getPersonName(person)}</p>
```

Comment:
Would like to be able to select/copy this text

Reply:
Hmm, I've made these buttons for accessibility reasons. Buttons have a default behavior which is to not allow text selection. IMO this is correct. If copying a user's name is important we should figure out the best place to put a copy user name button. This could be a right click menu or in the user's profile.

Reply:
Having this row be a button seems good to me -- it makes keyboard navigation easier and allows screen readers to parse it correctly. Does making the text in the button selectable with a `user-select: text;` css rule impact its accessibility? Keyboard navigation / button behavior seems to remain in tact in [this example](https://stackoverflow.com/questions/47090073/how-will-make-the-button-text-selectable).

Reply:
It lets you select the text but it still triggers the button press. So you can't really copy the text. I think this is a relatively minor thing that we can tackle later.